extension/oauth2clientauthextension/config.go

```diff
@@ -35,6 +37,43 @@ type Config struct {
 	// ClientSecretFile is the file pathg to read the application's secret from.
 	ClientSecretFile string `mapstructure:"client_secret_file"`
 
+	// ClientCertificateKeyID is the Key ID to include in the jwt. Only used if
+	// GrantType is set to "urn:ietf:params:oauth:grant-type:jwt-bearer".
```

Comment:
do we error out if ClientCertificateKeyID is set and GrantType is not "urn:ietf:params:oauth:grant-type:jwt-bearer" ?

Reply:
It's likely we can leave it as it is for a while. Let's see when people bump into this and complain they expect a different behavior.